helix-core/src/test/java/org/apache/helix/integration/TestZkCallbackHandlerLeak.java

```diff
@@ -467,23 +464,31 @@ public void testCurrentStatePathLeakingByAsycRemoval() throws Exception {
     cs.setSessionId(jobSessionId);
     cs.setStateModelDefRef(db0.getStateModelDefRef());
 
+    Map<String, List<String>> rpWatchPaths = ZkTestHelper.getZkWatch(rpManager.getZkClient());
+    Assert.assertFalse(rpWatchPaths.get("dataWatches").contains(jobKey.getPath()));
+
     LOG.info("add job");
-    boolean rtJob = false;
     for (int i = 0; i < mJobUpdateCnt; i++) {
-      rtJob = jobAccesor.setProperty(jobKey, cs);
+      jobAccesor.setProperty(jobKey, cs);
     }
 
+    // verify new watcher is installed on the new node
+    Thread.sleep(5000);
```

Comment:
I think we try to get rid of this kind of thread sleep, and use verifier. But as this is legacy test, I think it's fine. You can let @kaisun2000 know.

Reply:
I am not sure if we have watcher or listener installation verifier. If not, may be larger change is needed.

Reply:
Thanks for offline discuss. Updated.